### PR TITLE
Split 2 times only.

### PR DESCRIPTION
**Pull request by hibooboo2**
rancher/rancher#2800

### CODE REVIEW

File: code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/LdapTokenCreator.java [hunk 1]
```diff
@@ -39,10 +39,7 @@ public Token getToken(ApiRequest request) {
             throw new ClientVisibleException(ResponseCodes.SERVICE_UNAVAILABLE, "LdapConfig", "LdapConfig is not Configured.", null);
         }
         String code = ObjectUtils.toString(requestBody.get(SecurityConstants.CODE));
-        String[] split = code.split(":");
-        if (split.length != 2) {
-            throw new ClientVisibleException(ResponseCodes.FORBIDDEN);
-        }
+        String[] split = code.split(":", 2);
```
Comment (cjellick) [inline code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/LdapTokenCreator.java]: Do you still want to check for length of two? You could end up with an ArrayIndexOutOfBounds on the next line.

Comment (hibooboo2) [inline code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/LdapTokenCreator.java]: Ahh you mean in the even there is a malformed code. Sure I can do that. :)

Comment (hibooboo2) [inline code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/LdapTokenCreator.java]: Added the check. See next line.

**Comment by cjellick** [comment]
LGTM

**Pull request merged**